integration/benchmark/follower.go

```diff
@@ -114,8 +114,21 @@ Loop:
 		default:
 		}
 
+		nextHeight := f.Height() + 1
+
+		hdr, err := f.client.GetLatestBlockHeader(f.ctx, true)
+		if err != nil {
+			f.logger.Error().Err(err).Msg("failed to get latest block header")
+			continue
+		}
+		if hdr.Height < nextHeight {
+			f.logger.Trace().Uint64("want", nextHeight).Uint64("got", hdr.Height).
+				Msg("expected block is not yet sealed")
+			continue
+		}
+
 		GetBlockByHeightTime := time.Now()
-		block, err := f.client.GetBlockByHeight(f.ctx, f.height+1)
+		block, err := f.client.GetBlockByHeight(f.ctx, nextHeight)
```

Comment:
I thought you fix this in some PR, but it may be good to wait for all the collections to become available before processing the collections (at line 138).  otherwise, you could be reprocessing the same collection multiple times.

Reply:
(fixed in #3078)

Reply:
rebased.